cpp/net/connection_pool.cc

```diff
@@ -29,21 +145,59 @@ const HostPortPair& ConnectionPool::Connection::other_end() const {
 }
 
 
+void ConnectionPool::Connection::ReleaseConnection() {
+  conn_.release();
+  // Do not null out ssl_, despite the fact that it's a dangling pointer now we
+  // need it in the d'tor since the value of the pointer itself is a key into a
+  // map which needs to be updated.
```

Comment:
With a method like this to wrap the `unique_ptr` a bit (maybe `conn_` could be made private, with an accessor, to avoid mistakes?), why not do the removal from `connections_by_ssl_` here (before releasing)?